TicketUtil.py

```diff
@@ -119,6 +120,207 @@ def close_requests_session(self):
+                        "alias" : "SomeAlias",
+                        "op_sys" : "All",
+                        "priority" : "P1",
+                        "rep_platform" : "All"}
+        :param options_dict: Key: Value pairs for updating fields in Bugzilla.
+        :return: params: A dictionary to pass in to the POST request containing ticket details.
+        """
+        # Create our parameters for creating the ticket.
+        params = {'fields': {}}
+        params['fields']['project'] = {'key': self.project_key}
+
+        params = {"product": str(self.project_key)}
+
+        # Iterate through our options and add them to the params dict.
+        params.update(options_dict)
+        return params
+
+    def create_requests_session(self):
+        """
+        Returns to the super class if the authentication method is kerberos.
+        Creates a Requests Session and authenticates to base API URL with authentication other then kerberos.
+        We're using a Session to persist cookies across all requests made from the Session instance.
+        :return s: Requests Session.
+        """
+        if self.auth != 'rest':
+            return super()
+
+        try:
+            s = requests.Session()
+            r = s.get(self.auth_url, params=self.credentials, verify=False)
+            r.raise_for_status()
+            resp = r.json()
+            self.token = resp['token']
+            logging.debug("Create requests session: Status Code: {0}".format(r.status_code))
+            logging.info("Successfully authenticated to {0} with token: {1}".format(self.ticketing_tool, self.token))
+            return s
+
+        # We log an error if authentication was not successful, because rest of the HTTP requests will not succeed.
+        except requests.RequestException as e:
+            logging.error("Error authenticating to {0}. No valid credentials were provided.".format(self.auth_url))
+            logging.error(e.args[0])
+
+
+    def create_ticket(self, params):
+        """
+        Tries to create the ticket through the ticketing tool's API.
+        Retrieves the ticket_id and creates the ticket_url.
+        :param params: The payload to send in the POST request.
+        :return:
+        """
+        # Attempt to create ticket.
+        try:
+
+            if self.token:
+                params['token'] = self.token
+
+            r = self.s.post(self.rest_url, params=params)
+
+            r.raise_for_status()
+            logging.debug("Create ticket: Status Code: {0}".format(r.status_code))
+
+
+            self.ticket_id = r.json()['id']
+            self.ticket_url = self.craft_ticket_url()
+            logging.info("Created ticket {0} - {1}".format(self.ticket_id, self.ticket_url))
+
+        # If ticket creation is not successful, log an error.
+        except requests.RequestException as e:
+            logging.error("Error creating ticket")
+            logging.error(e.args[0])
+
```

Comment:
Remove this line. There should only be one blank line between methods inside a class.